# EBAY'S USED CAR LISTING ANALYSIS



In [9]:
import pandas as pd
import numpy as np

autos=pd.read_csv('autos.csv',encoding='Latin-1')


In [10]:
autos.head()

dateCrawled                                               name  seller  \
0  3/26/2016 17:47                   Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1   4/4/2016 13:38         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2  3/26/2016 18:57                         Volkswagen_Golf_1.6_United  privat   
3  3/12/2016 16:58  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...  privat   
4   4/1/2016 14:38  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...  privat   

  offerType    price   abtest vehicleType  yearOfRegistration    gearbox  \
0   Angebot  $5,000   control         bus                2004    manuell   
1   Angebot  $8,500   control   limousine                1997  automatik   
2   Angebot  $8,990      test   limousine                2009    manuell   
3   Angebot  $4,350   control  kleinwagen                2007  automatik   
4   Angebot  $1,350      test       kombi                2003    manuell   

   powerPS   model   odometer  monthOfRegistration fuelType       brand  \
0      158  andere  150,000km                    3      lpg     peugeot   
1      286     7er  150,000km                    6   benzin         bmw   
2      102    golf   70,000km                    7   benzin  volkswagen   
3       71  fortwo   70,000km                    6   benzin       smart   
4        0   focus  150,000km                    7   benzin        ford   

  notRepairedDamage     dateCreated  nrOfPictures  postalCode        lastSeen  
0              nein  3/26/2016 0:00             0       79588   4/6/2016 6:45  
1              nein   4/4/2016 0:00             0       71034  4/6/2016 14:45  
2              nein  3/26/2016 0:00             0       35394  4/6/2016 20:15  
3              nein  3/12/2016 0:00             0       33729  3/15/2016 3:16  
4              nein   4/1/2016 0:00             0       39218  4/1/2016 14:38

In [11]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

**There is 20 columns and 50000 rows in autos dataframe. Most of the cells are filled with information, only 'vehicle type', 'gearbox', 'model','fuelType','notRepairedDamage' columns contain some NaN values. 5 out of 20 columns contain numeric values. **

In [12]:
autos.columns
column_names=['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'registration_year', 'gearbox', 'powerPS', 'model',
       'odometer', 'registration_month', 'fuelType', 'brand',
       'unrepaired_damage', 'ad_created', 'nrOfPictures', 'postalCode',
       'lastSeen']

col_names=[]

for name in column_names:
    name=name.lower()
    col_names.append(name)

autos.columns=col_names
autos.columns


Index(['datecrawled', 'name', 'seller', 'offertype', 'price', 'abtest',
       'vehicletype', 'registration_year', 'gearbox', 'powerps', 'model',
       'odometer', 'registration_month', 'fueltype', 'brand',
       'unrepaired_damage', 'ad_created', 'nrofpictures', 'postalcode',
       'lastseen'],
      dtype='object')

**column names in the cell above has been slightly changed and lowercased in order to simplify further work**

****

# Initial Data Exploration and Cleaning

In [13]:
autos.describe(include='all')

datecrawled         name  seller offertype  price abtest  \
count             50000        50000   50000     50000  50000  50000   
unique            12073        38754       2         2   2357      2   
top     3/28/2016 14:49  Ford_Fiesta  privat   Angebot    $0    test   
freq                 15           78   49999     49999   1421  25756   
mean                NaN          NaN     NaN       NaN    NaN    NaN   
std                 NaN          NaN     NaN       NaN    NaN    NaN   
min                 NaN          NaN     NaN       NaN    NaN    NaN   
25%                 NaN          NaN     NaN       NaN    NaN    NaN   
50%                 NaN          NaN     NaN       NaN    NaN    NaN   
75%                 NaN          NaN     NaN       NaN    NaN    NaN   
max                 NaN          NaN     NaN       NaN    NaN    NaN   

       vehicletype  registration_year  gearbox       powerps  model  \
count        44905       50000.000000    47320  50000.000000  47242   
unique           8                NaN        2           NaN    245   
top      limousine                NaN  manuell           NaN   golf   
freq         12859                NaN    36993           NaN   4024   
mean           NaN        2005.073280      NaN    116.355920    NaN   
std            NaN         105.712813      NaN    209.216627    NaN   
min            NaN        1000.000000      NaN      0.000000    NaN   
25%            NaN        1999.000000      NaN     70.000000    NaN   
50%            NaN        2003.000000      NaN    105.000000    NaN   
75%            NaN        2008.000000      NaN    150.000000    NaN   
max            NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fueltype       brand unrepaired_damage  \
count       50000        50000.000000    45518       50000             40171   
unique         13                 NaN        7          40                 2   
top     150,000km                 NaN   benzin  volkswagen              nein   
freq        32424                 NaN    30107       10687             35232   
mean          NaN            5.723360      NaN         NaN               NaN   
std           NaN            3.711984      NaN         NaN               NaN   
min           NaN            0.000000      NaN         NaN               NaN   
25%           NaN            3.000000      NaN         NaN               NaN   
50%           NaN            6.000000      NaN         NaN               NaN   
75%           NaN            9.000000      NaN         NaN               NaN   
max           NaN           12.000000      NaN         NaN               NaN   

           ad_created  nrofpictures    postalcode       lastseen  
count           50000       50000.0  50000.000000          50000  
unique             76           NaN           NaN          10432  
top     4/3/2016 0:00           NaN           NaN  4/6/2016 2:45  
freq             1946           NaN           NaN            112  
mean              NaN           0.0  50813.627300            NaN  
std               NaN           0.0  25779.747957            NaN  
min               NaN           0.0   1067.000000            NaN  
25%               NaN           0.0  30451.000000            NaN  
50%               NaN           0.0  49577.000000            NaN  
75%               NaN           0.0  71540.000000            NaN  
max               NaN           0.0  99998.000000            NaN

* Columns that have mostly one value are candidates need to be dropped: 'seller', 'offertype'
* Columns that need more investigation: 'nrofpictures'
* Examples of numeric data stored as text need to be cleaned: 'price' and 'odometer' columns

In [14]:
autos['nrofpictures'].value_counts()

0    50000
Name: nrofpictures, dtype: int64

**It looks like the 'nrofphotos' column has 0 for every column. It needs to be removed along with the other two columns 'seller' and 'offertype'**

In [15]:
cols_to_drop=['nrofpictures','seller','offertype']
autos_dropped=autos.drop(cols_to_drop,axis=1)
autos_dropped.head()


datecrawled                                               name  \
0  3/26/2016 17:47                   Peugeot_807_160_NAVTECH_ON_BOARD   
1   4/4/2016 13:38         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  3/26/2016 18:57                         Volkswagen_Golf_1.6_United   
3  3/12/2016 16:58  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4   4/1/2016 14:38  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

     price   abtest vehicletype  registration_year    gearbox  powerps  \
0  $5,000   control         bus               2004    manuell      158   
1  $8,500   control   limousine               1997  automatik      286   
2  $8,990      test   limousine               2009    manuell      102   
3  $4,350   control  kleinwagen               2007  automatik       71   
4  $1,350      test       kombi               2003    manuell        0   

    model   odometer  registration_month fueltype       brand  \
0  andere  150,000km                   3      lpg     peugeot   
1     7er  150,000km                   6   benzin         bmw   
2    golf   70,000km                   7   benzin  volkswagen   
3  fortwo   70,000km                   6   benzin       smart   
4   focus  150,000km                   7   benzin        ford   

  unrepaired_damage      ad_created  postalcode        lastseen  
0              nein  3/26/2016 0:00       79588   4/6/2016 6:45  
1              nein   4/4/2016 0:00       71034  4/6/2016 14:45  
2              nein  3/26/2016 0:00       35394  4/6/2016 20:15  
3              nein  3/12/2016 0:00       33729  3/15/2016 3:16  
4              nein   4/1/2016 0:00       39218  4/1/2016 14:38

In [16]:
autos_dropped['price']=autos_dropped['price'].str.replace('$','').str.replace(',','').astype(int)

In [17]:
autos_dropped['odometer']=autos_dropped['odometer'].str.replace('km','').str.replace(',','').astype(int)

In [18]:
autos_dropped.rename({'odometer':'odometer_km'},axis=1,inplace=True)
autos_dropped.rename({'price':'price_$'},axis=1,inplace=True)
autos_dropped.head()

datecrawled                                               name  \
0  3/26/2016 17:47                   Peugeot_807_160_NAVTECH_ON_BOARD   
1   4/4/2016 13:38         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  3/26/2016 18:57                         Volkswagen_Golf_1.6_United   
3  3/12/2016 16:58  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4   4/1/2016 14:38  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   price_$   abtest vehicletype  registration_year    gearbox  powerps  \
0     5000  control         bus               2004    manuell      158   
1     8500  control   limousine               1997  automatik      286   
2     8990     test   limousine               2009    manuell      102   
3     4350  control  kleinwagen               2007  automatik       71   
4     1350     test       kombi               2003    manuell        0   

    model  odometer_km  registration_month fueltype       brand  \
0  andere       150000                   3      lpg     peugeot   
1     7er       150000                   6   benzin         bmw   
2    golf        70000                   7   benzin  volkswagen   
3  fortwo        70000                   6   benzin       smart   
4   focus       150000                   7   benzin        ford   

  unrepaired_damage      ad_created  postalcode        lastseen  
0              nein  3/26/2016 0:00       79588   4/6/2016 6:45  
1              nein   4/4/2016 0:00       71034  4/6/2016 14:45  
2              nein  3/26/2016 0:00       35394  4/6/2016 20:15  
3              nein  3/12/2016 0:00       33729  3/15/2016 3:16  
4              nein   4/1/2016 0:00       39218  4/1/2016 14:38

**further exploring of 'price_$' and 'odometer' columns:**

In [19]:
autos_dropped['price_$'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price_$, dtype: float64

In [20]:
autos_dropped['price_$'].unique().shape

(2357,)

In [21]:
autos_dropped['price_$'].value_counts().sort_index(ascending=True).head()

0    1421
1     156
2       3
3       1
5       2
Name: price_$, dtype: int64

In [22]:
#autos_dropped=autos_dropped[autos_dropped['price_$'].between(1,351000)]  more simple way to put that
autos_dropped=autos_dropped[(autos_dropped['price_$']>1) & (autos_dropped['price_$']<351000)]
autos_dropped['price_$'].describe()

count     48409.000000
mean       5907.909707
std        9068.263463
min           2.000000
25%        1250.000000
50%        3000.000000
75%        7490.000000
max      350000.000000
Name: price_$, dtype: float64

In [23]:
autos_dropped['odometer_km'].value_counts()

150000    31307
125000     5046
100000     2108
90000      1733
80000      1414
70000      1215
60000      1154
50000      1011
40000       815
5000        815
30000       779
20000       762
10000       250
Name: odometer_km, dtype: int64

**We can see that the values in this field are rounded, which might indicate that sellers had to choose from pre-set options for this field. Additionally, there are more high mileage than low mileage vehicles.**

# Exploring the date columns
**There are a number of columns with date information:**

* date_crawled
* registration_month
* registration_year
* ad_created
* last_seen

These are a combination of dates that were crawled, and dates with meta-information from the crawler. The non-registration dates are stored as strings.

In [24]:
datecrawled=autos_dropped['datecrawled'].str[:10]
datecrawled

0        3/26/2016 
1        4/4/2016 1
2        3/26/2016 
3        3/12/2016 
4        4/1/2016 1
5        3/21/2016 
6        3/20/2016 
7        3/16/2016 
8        3/22/2016 
9        3/16/2016 
10       3/15/2016 
11       3/16/2016 
12       3/31/2016 
13       3/23/2016 
14       3/23/2016 
15       4/1/2016 1
16       3/16/2016 
17       3/29/2016 
18       3/26/2016 
19       3/17/2016 
20       3/5/2016 1
21       3/6/2016 1
22       3/28/2016 
23       3/10/2016 
24       4/3/2016 1
25       3/21/2016 
26       4/3/2016 2
28       3/19/2016 
29       4/2/2016 1
30       3/14/2016 
            ...    
49968    4/1/2016 1
49969    3/17/2016 
49970    3/21/2016 
49971    3/29/2016 
49972    3/26/2016 
49973    3/27/2016 
49975    3/27/2016 
49976    3/19/2016 
49977    3/31/2016 
49978    4/4/2016 1
49979    3/20/2016 
49980    3/12/2016 
49981    3/15/2016 
49982    3/29/2016 
49983    3/6/2016 1
49985    4/2/2016 1
49986    4/4/2016 2
49987    3/22/2016 
49988    3/28/2016 


In [25]:
datecrawled.value_counts(normalize=True,dropna=False).sort_index(ascending=False)

4/7/2016 9    0.000083
4/7/2016 8    0.000083
4/7/2016 7    0.000083
4/7/2016 6    0.000021
4/7/2016 4    0.000021
4/7/2016 3    0.000083
4/7/2016 2    0.000021
4/7/2016 1    0.000806
4/7/2016 0    0.000186
4/6/2016 9    0.000062
4/6/2016 8    0.000124
4/6/2016 7    0.000083
4/6/2016 6    0.000103
4/6/2016 5    0.000062
4/6/2016 3    0.000062
4/6/2016 2    0.000682
4/6/2016 1    0.001859
4/6/2016 0    0.000124
4/5/2016 9    0.001487
4/5/2016 8    0.000640
4/5/2016 7    0.000764
4/5/2016 6    0.000186
4/5/2016 5    0.000083
4/5/2016 4    0.000083
4/5/2016 3    0.000124
4/5/2016 2    0.000971
4/5/2016 1    0.007251
4/5/2016 0    0.001487
4/4/2016 9    0.001074
4/4/2016 8    0.000764
                ...   
3/6/2016 5    0.000021
3/6/2016 4    0.000145
3/6/2016 3    0.000103
3/6/2016 2    0.003140
3/6/2016 1    0.008821
3/6/2016 0    0.000785
3/5/2016 2    0.006610
3/5/2016 1    0.018757
3/31/2016     0.031812
3/30/2016     0.033733
3/29/2016     0.034126
3/28/2016     0.034849
3/27/2016  

**Looks like the site was crawled daily over a one month period in March and April 2016.**

In [26]:
autos_dropped['ad_created'].str[:10].value_counts().sort_index()

1/10/2016        2
1/13/2016        1
1/14/2016        1
1/16/2016        1
1/22/2016        1
1/27/2016        3
1/29/2016        1
1/3/2016 0       1
1/7/2016 0       1
11/10/2015       1
12/30/2015       1
12/5/2015        1
2/1/2016 0       1
2/11/2016        1
2/12/2016        2
2/14/2016        2
2/16/2016        1
2/17/2016        1
2/18/2016        2
2/19/2016        3
2/2/2016 0       2
2/20/2016        2
2/21/2016        3
2/22/2016        1
2/23/2016        4
2/24/2016        2
2/25/2016        3
2/26/2016        2
2/27/2016        6
2/28/2016       10
              ... 
3/2/2016 0       5
3/20/2016     1833
3/21/2016     1817
3/22/2016     1585
3/23/2016     1554
3/24/2016     1420
3/25/2016     1534
3/26/2016     1564
3/27/2016     1503
3/28/2016     1693
3/29/2016     1649
3/3/2016 0      42
3/30/2016     1624
3/31/2016     1542
3/4/2016 0      72
3/5/2016 0    1110
3/6/2016 0     742
3/7/2016 0    1684
3/8/2016 0    1612
3/9/2016 0    1603
4/1/2016 0    1632
4/2/2016 0  

**There is a large variety of ad created dates. Most fall within 1-2 months of the listing date, but a few are quite old, with the oldest at around 9 months.**

# Dealing with Incorrect Registration Year Data

In [27]:
autos_dropped["registration_year"].describe()

count    48409.000000
mean      2004.774319
std         88.783278
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

**The year that the car was first registered will likely indicate the age of the car. Looking at this column, we note some odd values. The minimum value is 1000, long before cars were invented and the maximum is 9999, many years into the future**

Because a car can't be first registered before the listing was seen, any vehicle with a registration year above 2016 is definitely inaccurate. Determining the earliest valid year is more difficult. Realistically, it could be somewhere in the first few decades of the 1900s.

In [28]:
autos_dropped['registration_year'].value_counts().sort_index()

1000       1
1001       1
1111       1
1800       2
1910       4
1927       1
1929       1
1931       1
1934       2
1937       4
1938       1
1939       1
1941       2
1943       1
1948       1
1950       2
1951       2
1952       1
1953       1
1954       2
1955       2
1956       4
1957       2
1958       4
1959       6
1960      22
1961       6
1962       4
1963       8
1964      12
        ... 
2000    3128
2001    2632
2002    2480
2003    2697
2004    2700
2005    2927
2006    2670
2007    2274
2008    2213
2009    2083
2010    1589
2011    1623
2012    1310
2013     802
2014     662
2015     391
2016    1209
2017    1386
2018     470
2019       2
2800       1
4100       1
4500       1
4800       1
5000       4
5911       1
6200       1
8888       1
9000       1
9999       3
Name: registration_year, Length: 95, dtype: int64

**Given that number of registrations earlier than 1900 and later that 2019 is less than 4% of our data, we will remove these rows.**

In [29]:
(~autos_dropped['registration_year'].between(1910,2016)).sum()/autos_dropped.shape[0]


0.03879443905058977

In [30]:
autos_dropped=autos_dropped[autos_dropped['registration_year'].between(1910,2016)]

In [31]:
autos_dropped['registration_year'].value_counts(normalize=True).head(10)

2000    0.067224
2005    0.062904
1999    0.062023
2004    0.058026
2003    0.057961
2006    0.057381
2001    0.056564
2002    0.053298
1998    0.050547
2007    0.048871
Name: registration_year, dtype: float64

**It looks like most of the vehicles were registered in the past 20 years**

In [32]:
autos_dropped['registration_year'].describe()

count    46531.000000
mean      2002.926049
std          7.159928
min       1910.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       2016.000000
Name: registration_year, dtype: float64

# Exploring car brands

In [33]:
autos_dropped['brand'].value_counts(normalize=True)

volkswagen        0.211343
bmw               0.109948
opel              0.107369
mercedes_benz     0.096452
audi              0.086609
ford              0.069975
renault           0.047130
peugeot           0.029873
fiat              0.025682
seat              0.018267
skoda             0.016419
nissan            0.015302
mazda             0.015237
smart             0.014206
citroen           0.014034
toyota            0.012744
hyundai           0.010036
sonstige_autos    0.009649
volvo             0.009134
mini              0.008790
mitsubishi        0.008188
honda             0.007844
kia               0.007092
alfa_romeo        0.006641
porsche           0.006103
suzuki            0.005932
chevrolet         0.005695
chrysler          0.003525
dacia             0.002643
daihatsu          0.002514
jeep              0.002278
subaru            0.002128
land_rover        0.002106
saab              0.001655
jaguar            0.001569
daewoo            0.001504
trabant           0.001397
r


**German manufacturers represent four out of the top five brands, almost 50% of the overall listings. Volkswagen is by far the most popular brand, with approximately double the cars for sale of the next two brands combined.**

**There are lots of brands that don't have a significant percentage of listings, so we will limit our analysis to top 20 brands.**

In [34]:
top20brands=autos_dropped['brand'].value_counts(normalize=True).head(20)
top20brands

volkswagen        0.211343
bmw               0.109948
opel              0.107369
mercedes_benz     0.096452
audi              0.086609
ford              0.069975
renault           0.047130
peugeot           0.029873
fiat              0.025682
seat              0.018267
skoda             0.016419
nissan            0.015302
mazda             0.015237
smart             0.014206
citroen           0.014034
toyota            0.012744
hyundai           0.010036
sonstige_autos    0.009649
volvo             0.009134
mini              0.008790
Name: brand, dtype: float64

# Mean price by brand calculation

In [35]:
brands=top20brands.index
mean_price_by_brand={}

for b in brands:
    selected_brands=autos_dropped[autos_dropped['brand']==b]
    brand_mean=selected_brands['price_$'].mean()
    mean_price_by_brand[b]=int(brand_mean)

mean_price_by_brand



{'volkswagen': 5417,
 'bmw': 8367,
 'opel': 2990,
 'mercedes_benz': 8657,
 'audi': 9362,
 'ford': 3757,
 'renault': 2483,
 'peugeot': 3100,
 'fiat': 2818,
 'seat': 4412,
 'skoda': 6384,
 'nissan': 4750,
 'mazda': 4112,
 'smart': 3580,
 'citroen': 3784,
 'toyota': 5167,
 'hyundai': 5376,
 'sonstige_autos': 12585,
 'volvo': 4969,
 'mini': 10613}

**Of the top 5 brands, there is a distinct price gap:**

**Audi, BMW and Mercedes Benz are more expensive.
Ford and Opel are less expensive.
Volkswagen is in between - this may explain its popularity, it may be a 'best of 'both worlds' option.**

# Exploring mileage

In [36]:
mean_milage_by_brand={}

for b in brands:
    selected_brands=autos_dropped[autos_dropped['brand']==b]
    mean_milage=selected_brands['odometer_km'].mean()
    mean_milage_by_brand[b]=mean_milage

mean_milage_by_brand

{'volkswagen': 128708.56213138092,
 'bmw': 132553.75293197812,
 'opel': 129322.4579663731,
 'mercedes_benz': 130838.9037433155,
 'audi': 129208.43672456576,
 'ford': 124210.68796068797,
 'renault': 128144.09484724123,
 'peugeot': 127122.30215827338,
 'fiat': 117066.94560669456,
 'seat': 121029.41176470589,
 'skoda': 110746.07329842931,
 'nissan': 118370.78651685393,
 'mazda': 124464.03385049365,
 'smart': 99326.77760968229,
 'citroen': 119647.77947932619,
 'toyota': 115944.35075885328,
 'hyundai': 106541.75588865097,
 'sonstige_autos': 90256.12472160357,
 'volvo': 138294.11764705883,
 'mini': 88105.13447432763}

In [37]:
brand_mean_price=pd.Series(mean_price_by_brand)
brand_mean_price

volkswagen         5417
bmw                8367
opel               2990
mercedes_benz      8657
audi               9362
ford               3757
renault            2483
peugeot            3100
fiat               2818
seat               4412
skoda              6384
nissan             4750
mazda              4112
smart              3580
citroen            3784
toyota             5167
hyundai            5376
sonstige_autos    12585
volvo              4969
mini              10613
dtype: int64

In [38]:
brand_mean_mileage=pd.Series(mean_milage_by_brand)
brand_mean_mileage

volkswagen        128708.562131
bmw               132553.752932
opel              129322.457966
mercedes_benz     130838.903743
audi              129208.436725
ford              124210.687961
renault           128144.094847
peugeot           127122.302158
fiat              117066.945607
seat              121029.411765
skoda             110746.073298
nissan            118370.786517
mazda             124464.033850
smart              99326.777610
citroen           119647.779479
toyota            115944.350759
hyundai           106541.755889
sonstige_autos     90256.124722
volvo             138294.117647
mini               88105.134474
dtype: float64

In [39]:
df=pd.DataFrame(brand_mean_price,columns=['mean_price'])
df

mean_price
volkswagen            5417
bmw                   8367
opel                  2990
mercedes_benz         8657
audi                  9362
ford                  3757
renault               2483
peugeot               3100
fiat                  2818
seat                  4412
skoda                 6384
nissan                4750
mazda                 4112
smart                 3580
citroen               3784
toyota                5167
hyundai               5376
sonstige_autos       12585
volvo                 4969
mini                 10613

In [40]:
df['mean_mileage']=brand_mean_mileage
df

mean_price   mean_mileage
volkswagen            5417  128708.562131
bmw                   8367  132553.752932
opel                  2990  129322.457966
mercedes_benz         8657  130838.903743
audi                  9362  129208.436725
ford                  3757  124210.687961
renault               2483  128144.094847
peugeot               3100  127122.302158
fiat                  2818  117066.945607
seat                  4412  121029.411765
skoda                 6384  110746.073298
nissan                4750  118370.786517
mazda                 4112  124464.033850
smart                 3580   99326.777610
citroen               3784  119647.779479
toyota                5167  115944.350759
hyundai               5376  106541.755889
sonstige_autos       12585   90256.124722
volvo                 4969  138294.117647
mini                 10613   88105.134474

**The range of car mileages does not vary as much as the prices do by brand, instead all falling within 10% for the top brands.**

**There is a slight trend to the more expensive vehicles having higher mileage, with the less expensive vehicles having lower mileage.**

**Calculating how much cheaper are cars with damage than their non-damaged counterparts:**

In [41]:
non_damaged=autos_dropped[autos_dropped['unrepaired_damage']=='nein']
damaged=autos_dropped[autos_dropped['unrepaired_damage']=='ja']

average_price_diff=non_damaged['price_$'].mean()-damaged['price_$'].mean()
average_price_diff

4921.314493438735